### PR TITLE
Main 6430

### PR DESCRIPTION
**Pull request by Archbold**

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/ArticlePageObject.java [hunk 1]
```diff
@@ -9,6 +9,8 @@
 import com.wikia.webdriver.common.dataprovider.VisualEditorDataProvider.Formatting;
 import com.wikia.webdriver.common.dataprovider.VisualEditorDataProvider.Style;
 import com.wikia.webdriver.common.logging.PageObjectLogging;
+import com.wikia.webdriver.elements.mercury.pages.ArticlePage;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/ArticlePageObject.java]: MINOR Remove this unused import 'com.wikia.webdriver.elements.mercury.pages.ArticlePage'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/testcases/mediatests/addvideo/VetArticleCommentsTests.java [hunk 2]
```diff
@@ -5,6 +5,7 @@
 
 import com.wikia.webdriver.common.contentpatterns.PageContent;
 import com.wikia.webdriver.common.contentpatterns.VideoContent;
+import com.wikia.webdriver.common.core.Assertion;
 import com.wikia.webdriver.common.core.annotations.Execute;
 import com.wikia.webdriver.common.core.annotations.RelatedIssue;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mediatests/addvideo/VetArticleCommentsTests.java]: MINOR Remove this unused import 'com.wikia.webdriver.common.core.annotations.RelatedIssue'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/elements/oasis/components/comment/ArticleComment.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  }
+
+  public boolean isVideoVisible(String videoName) {
+    try {
+      WebElement
+          element =
+          driver.findElement(
+              By.cssSelector(videoInCommentsSelector.replace("%videoName%", videoName)));
+      return element.isDisplayed();
+    } catch (ElementNotFoundException e) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/oasis/components/comment/ArticleComment.java]: MAJOR Either log or rethrow this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues:
* MAJOR 1 major
* MINOR 2 minor

Watch the comments in this conversation to review them.

File: src/test/java/com/wikia/webdriver/elements/oasis/components/comment/ArticleComment.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+import com.gargoylesoftware.htmlunit.ElementNotFoundException;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+public class ArticleComment extends BasePageObject {
+
+  @FindBy(css = "#article-comm-submit")
+  protected WebElement commentSubmitButton;
```
Comment (nikodamn) [inline src/test/java/com/wikia/webdriver/elements/oasis/components/comment/ArticleComment.java]: Is there any strong reason to make this variable protected not private?

File: src/test/java/com/wikia/webdriver/elements/oasis/components/comment/ArticleComment.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  String videoInCommentsSelector = ".speech-bubble-message img[data-video-name*='%videoName%']";
+
```
Comment (nikodamn) [inline src/test/java/com/wikia/webdriver/elements/oasis/components/comment/ArticleComment.java]: Please remove this extra new line

File: src/test/java/com/wikia/webdriver/elements/oasis/components/comment/ArticleComment.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    scrollAndClick(commentSubmitButton);
+    waitForElementNotVisibleByElement(commentSubmitButton, 30);
+    PageObjectLogging.log("submitComment", "comment has been submitted", true);
+
+    return this;
+  }
+
+  public boolean isVideoVisible(String videoName) {
+    try {
+      WebElement
```
Comment (nikodamn) [inline src/test/java/com/wikia/webdriver/elements/oasis/components/comment/ArticleComment.java]: This line would look nicer that way: "WebElement element ="

**Comment by nikodamn** [comment]
@Archbold  CR is done

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/ArticlePageObject.java (hunk 1 above)
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/ArticlePageObject.java]: MINOR Remove this unused import 'com.wikia.webdriver.elements.mercury.pages.ArticlePage'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/testcases/mediatests/addvideo/VetArticleCommentsTests.java (hunk 2 above)
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mediatests/addvideo/VetArticleCommentsTests.java]: MINOR Remove this unused import 'com.wikia.webdriver.common.core.annotations.RelatedIssue'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/elements/oasis/components/comment/ArticleComment.java [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+          element =
+          driver.findElement(
+              By.cssSelector(videoInCommentsSelector.replace("%videoName%", videoName)));
+      return element.isDisplayed();
+    } catch (ElementNotFoundException e) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/oasis/components/comment/ArticleComment.java]: MAJOR Either log or rethrow this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues:
* MAJOR 1 major
* MINOR 2 minor

Watch the comments in this conversation to review them.

File: src/test/java/com/wikia/webdriver/elements/oasis/components/comment/ArticleComment.java (hunk 5 above)
Comment (Archbold) [inline src/test/java/com/wikia/webdriver/elements/oasis/components/comment/ArticleComment.java]: Extra new line removed

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues:
* MAJOR 1 major
* MINOR 2 minor

Watch the comments in this conversation to review them.

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues:
* MAJOR 1 major
* MINOR 2 minor

Watch the comments in this conversation to review them.

File: src/test/java/com/wikia/webdriver/elements/oasis/components/comment/ArticleComment.java (hunk 4 above)
Comment (ludwikkazmierczak) [inline src/test/java/com/wikia/webdriver/elements/oasis/components/comment/ArticleComment.java]: No need for this to be protected +1

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues:
* MAJOR 1 major
* MINOR 2 minor

Watch the comments in this conversation to review them.

File: src/test/java/com/wikia/webdriver/elements/oasis/components/comment/ArticleComment.java [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
+public class ArticleComment extends BasePageObject {
+
+  @FindBy(css = "#article-comm-submit")
+  protected WebElement commentSubmitButton;
+  String videoInCommentsSelector = ".speech-bubble-message img[data-video-name*='%videoName%']";
+
+
+  public ArticleComment waitForVideo() {
+    driver.switchTo().frame(1);
```
Comment (Archbold) [inline src/test/java/com/wikia/webdriver/elements/oasis/components/comment/ArticleComment.java]: Done

File: src/test/java/com/wikia/webdriver/elements/oasis/components/comment/ArticleComment.java (hunk 6 above)
Comment (Archbold) [inline src/test/java/com/wikia/webdriver/elements/oasis/components/comment/ArticleComment.java]: Done

**Comment by Archbold** [comment]
Changes made after code review. Please CR again.

**Comment by ludwikkazmierczak** [comment]
LGTM

**Pull request merged**